rorapi/middleware.py

```diff
@@ -0,0 +1,43 @@
+from django.core.cache import cache
+from django.http import JsonResponse
+from rorapi.v2.models import Client
+from django.utils.timezone import now
+import os
+
+# Toggle Behavior-Based Rate Limiting
```

Comment:
Rate limiting should ideally be applied at the load balancer rather than at the application level. See some implementation options in this spec https://docs.google.com/document/d/1FhI6wFHty6R4NauKapYOzJtaikaAHYGtfgwWxxdtrhw